lib/handlers/copy.js

```diff
@@ -0,0 +1,76 @@
+module.exports = handler
+
+const utils = require('../utils')
+const mkdirp = require('fs-extra').mkdirp
+const fs = require('fs')
+const request = require('request')
+const path = require('path')
+const debug = require('../debug')
+const error = require('../http-error')
+
+/**
+ * Cleans up a file write stream (ends stream, deletes the file).
+ * @method cleanupFileStream
+ * @param stream {WriteStream}
+ */
+function cleanupFileStream (stream) {
+  let streamPath = stream.path
+  stream.destroy()
+  fs.unlinkSync(streamPath)
+}
+
+/**
+ * Handles HTTP COPY requests to import a given resource (specified in the
+ * `Source:` header) to a destination (specified in request path).
+ * For the moment, you can copy from public resources only (no auth delegation
+ * is implemented), and is mainly intended for use with
+ * "Save an external resource to Solid" type apps.
+ * @method handler
+ */
+function handler (req, res, next) {
```

Comment:
hey @dmitrizagidulin in order to make our life easy once we want to move to another type of store, we are abstracting all of the file system stuff into ldp.js. The way I see it is the following:

1) File system
2) LDP.copy
3) HTTP Server handling

this `copy.js` should call LDP.copy (that will be implemented with fs calls and a callback*) and in the callback* do the http handling (see get.js for example!)